docs/usage/guide_f5.rst

```diff
@@ -0,0 +1,316 @@
+
+
+Common parameters
+`````````````````
+
+Every F5 module accepts the following parameters:
+
+server
+   Host name or IP address of BIG-IP.
+
+user
+   The user who can connect to BIG-IP. This user must have administrative privileges.
+
+password
+   Password for the user. Use Ansible Vault to encrypt the password, rather than storing it as plain text in your playbook or inventory file.
+
+validate_certs
+   Use to validate self-signed SSL certificates on personally-controlled sites.
+
+
+Common Tasks
+````````````
+
+Use the F5 Modules for Ansible to perform actions against BIG-IP.
+
+For reference information for each module, see `this list <http://docs.ansible.com/ansible/latest/list_of_network_modules.html#f5>`_.
+
+License BIG-IP
+++++++++++++++
+
+If you have a BIG-IP license, you can use Ansible to license BIG-IP. This example shows the full playbook.
+
+.. code-block:: yaml
+
+   ---
+
+   - name: License BIG-IP
+     hosts: f5-test
+     connection: local
+
+     vars:
+         bigip_license: "XXXXX-XXXXX-XXXXX-XXXXX-XXXXXXX"
+
+     tasks:
+         - name: License BIG-IP
+           bigip_license:
+               server: "{{ ansible_host }}"
+               user: "{{ bigip_username }}"
+               password: "{{ bigip_password }}"
+               validate_certs: "{{ validate_certs }}"
+               server_port: "{{ bigip_port }}"
+               key: "{{ bigip_license }}"
+           delegate_to: localhost
+
+
+Provision BIG-IP
+++++++++++++++++
+
+Then you can use Ansible to provision BIG-IP modules.
+
+.. code-block:: yaml
+
+     tasks:
+         - name: Provision ASM at "nominal" level
+           bigip_provision:
+               server: "{{ ansible_host }}"
+               user: "{{ bigip_username }}"
+               password: "{{ bigip_password }}"
+               validate_certs: "{{ validate_certs }}"
+               module: asm
+               level: nominal
+           delegate_to: localhost
+
+For more ideas on how you might use Ansible for initial BIG-IP setup `see this doc <https://devcentral.f5.com/codeshare/automate-f5-initial-setup-icontrol-amp-ansible-930>`_.
+
+Create pool members, a pool, and a virtual server
++++++++++++++++++++++++++++++++++++++++++++++++++
+
+You can use the F5 Modules for Ansible to create a pool and add members to it, and to add the pool to the virtual server.
+
+For a full walkthrough of this example, `see this doc <http://clouddocs.f5.com/products/orchestration/ansible/devel/usage/playbook_tutorial.html>`_.
+
+.. code-block:: yaml
+
+     tasks:
+         - name: Create a pool
+           bigip_pool:
+               server: "{{ ansible_host }}"
+                  user: "{{ bigip_username }}"
+               password: "{{ bigip_password }}"
+               validate_certs: "{{ validate_certs }}"
+               lb_method: "ratio-member"
+               name: "web_pool"
+               slow_ramp_time: "120"
+           delegate_to: localhost
+
+         - name: Create node1
```

Comment:
this can be removed. Nodes are created automatically when pool members are added.